test/gurka/gurka.h

```diff
@@ -99,6 +99,47 @@ void build_pbf(const nodelayout& node_locations,
 std::vector<std::vector<std::string>> get_paths(const valhalla::Api& result);
 } // namespace detail
 
+/**
+ * Generate a new GraphReader that doesn't re-use a previously
+ * statically initizalized tile_extract member variable.
+ *
+ * Useful if you need to reload a tile extract within the same
+ * process
+ */
+std::shared_ptr<valhalla::baldr::GraphReader>
+make_clean_graphreader(const boost::property_tree::ptree& mjolnir_conf);
+
+/*************************************************************/
+// Creates an empty traffic file
+//
+// To actually customize the traffic data, use `customize_live_traffic_data`
+//
+/*************************************************************/
+void build_live_traffic_data(const boost::property_tree::ptree& config,
+                             uint32_t traffic_tile_version = valhalla::baldr::TRAFFIC_TILE_VERSION);
+
+/*************************************************************/
+// Helper function for customizing traffic data in unit-tests
+//
+// `setter_cb` is a callback that can modify traffic for each edge
+// when building traffic data
+/*************************************************************/
+using LiveTrafficCustomize = std::function<void(valhalla::baldr::GraphReader&,
+                                                valhalla::baldr::TrafficTile&,
+                                                int,
+                                                valhalla::baldr::TrafficSpeed*)>;
+
+void customize_live_traffic_data(const boost::property_tree::ptree& config,
+                                 const LiveTrafficCustomize& setter_cb);
+
+using HistoricalTrafficCustomize =
+    std::function<boost::optional<std::array<float, baldr::kBucketsPerWeek>>(baldr::DirectedEdge&)>;
+void customize_historical_traffic(const boost::property_tree::ptree& config,
+                                  const HistoricalTrafficCustomize& cb);
+
+using EdgesCustomize = std::function<void(const baldr::GraphId&, baldr::DirectedEdge&)>;
+void customize_edges(const boost::property_tree::ptree& config, const EdgesCustomize& setter_cb);
+
```

Comment:
this stuff should go back to the file it came from and the rest of those changes reverted. if you want to you can combine gurka sources and test sources into the single test target